src/main/java/core/basesyntax/operation/BalanceHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class BalanceHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;

```

Comment:
You should not use a concrete implementation (`ShopServiceImpl`) here. Instead, use an interface to follow the Dependency Inversion Principle of SOLID. It's one of the ways to achieve loose coupling between classes.